tsdb/head_append.go

```diff
@@ -274,7 +274,7 @@ func (a *headAppender) Append(ref storage.SeriesRef, lset labels.Labels, t int64
 	}
 
 	s.Lock()
-	if err := s.appendable(t, v); err != nil {
+	if err := s.appendable(t, v); err != nil || s.deleted {
```

Comment:
This will silently discard the sample because we will return a nil error. And in that case Cortex will not return any error and Prometheus will never send that sample again and the sample will be lost. We should be returning an error here such that remote write also retries it.

Reply:
yeah.. i will update the PR with something like :https://github.com/alanprot/prometheus/commit/2d355656e922f3578ba0af0321e1b21113eb0171

Reply:
I updated the PR! :D